synapse/push/mailer.py

```diff
@@ -890,7 +890,7 @@ def safe_markup(raw_html: str) -> Markup:
                 tags=ALLOWED_TAGS,
                 attributes=ALLOWED_ATTRS,
                 # bleach master has this, but it isn't released yet
-                # protocols=ALLOWED_SCHEMES,
+                protocols=ALLOWED_SCHEMES,
```

Comment:
We're using [`bleach == 6.0.0` in `poetry.lock`](https://github.com/matrix-org/synapse/blob/4ba528d9c31e1af75f3532ac2a52c1b12365f592/poetry.lock#L203-L204) but it looks like there is a [breaking change where it needs to be a `Set` now](https://github.com/mozilla/bleach/blob/13d6c0c91d2b1bef52ad2b5a2812f76dda18298f/CHANGES#L9-L36). We can't really support both versions before and after as far as I can tell so would need to converge either way on our dependency versions

Reply:
That's slightly alarming. I couldn't see a way in which this was obviously broken if you pass in a list:

```
$ python
Python 3.11.3 (main, May 24 2023, 00:00:00) [GCC 12.3.1 20230508 (Red Hat 12.3.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import bleach
>>> bleach.__version__
'6.0.0'
>>> bleach.clean()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: clean() missing 1 required positional argument: 'text'
>>> bleach.clean('aaaaa')
'aaaaa'
>>> bleach.clean('<naughty>')
'&lt;naughty&gt;'
>>> bleach.clean('<naughty><a>')
'&lt;naughty&gt;<a></a>'
>>> bleach.clean('<naughty><a href="foo">')
'&lt;naughty&gt;<a href="foo"></a>'
>>> bleach.clean('<naughty><a href="foo">', tags=['a'])
'&lt;naughty&gt;<a href="foo"></a>'
>>> bleach.clean('<naughty><a href="foo">', tags={'a'})
'&lt;naughty&gt;<a href="foo"></a>'
>>> bleach.clean('<naughty><a href="foo">', tags={'a', 'naughty'})
'<naughty><a href="foo"></a></naughty>'
>>> bleach.clean('<naughty><a href="foo">', tags={'naughty'})
'<naughty>&lt;a href="foo"&gt;</naughty>'
>>> bleach.clean('<naughty><a href="foo">', tags=['naughty'])
'<naughty>&lt;a href="foo"&gt;</naughty>'
```